aten/src/THC/THCTensorRandom.cpp

```diff
@@ -74,18 +77,21 @@ static THCGenerator* THCRandom_rawGenerator(THCState* state)
 THCGenerator* THCRandom_getGenerator(THCState* state)
 {
   THCGenerator* gen = THCRandom_rawGenerator(state);
-  if (gen->initf == 0)
+  std::lock_guard<std::mutex> lock(gen->mutex);
+  if (gen->gen_state.initf == 0)
   {
     initializeGenerator(state, gen);
-    createGeneratorState(gen, gen->initial_seed);
-    gen->initf = 1;
+    createGeneratorState(gen, gen->gen_state.initial_seed);
+    gen->gen_state.initf = 1;
   }
   return gen;
 }
 
 struct curandStateMtgp32* THCRandom_generatorStates(struct THCState* state)
 {
-  return THCRandom_getGenerator(state)->gen_states;
+  THCGenerator* gen = THCRandom_getGenerator(state);
+  std::lock_guard<std::mutex> lock(gen->mutex);
```

Comment:
I can't see how is this lock supposed to help. It calls `getGenerator` which should ensure that it's ready to be used by now, and this doesn't protect the actual use of the states anyway